### ISSUE TITLE
Windows installation requires all-purpose root CA trusted certificate?

### ISSUE THREAD

**Issue by Terrance** [UnicornsOnLSD/finamp#1487]
When opening the latest 0.9.21 MSIX package for installation, Windows 10 blocks it:

<img width="652" height="535" alt="Image" src="https://github.com/user-attachments/assets/76b2770f-c47d-435c-afac-1249b4db0341" />

> This app package’s publisher certificate could not be verified. Contact your system administrator or the app developer to obtain a new app package with verified certificates. The root certificate and all immediate certificates of the signature in the app package must be verified (0x800B010A)

I can work around this by extracting the signing certificate from Properties > Digital Signatures, and adding it as a machine-wide trusted CA certificate, but this is far from ideal (notably, the certificate is valid for all purposes, rather than just code-signing).  Is there a better way to handle this?

**Comment by Chaphasilor** [comment]
Yeah so for that we need a Microsoft-trusted cert (for msix), and we get that once we set up Finamp on the Microsoft Store. That will definitely happen at the start of next year, but until then you can install the custom certificate for Finamp like you did, or by following the instructions here:  
https://github.com/jmshrv/finamp/tree/new-readme?tab=readme-ov-file#windows

**Comment by Terrance** [comment]
Whilst I can probably trust you to maintain the security of your signing certificate, I really don't want the liability of having a custom root CA present with the potential ability to spoof certificates for general TLS/HTTPS connections.

I added the certificate temporarily for an initial install and dropped it soon after, but readding/removing it for every update is a bit of a nuisance.  I could unpack the ZIP, though it means losing out on having things consistently included in the Start menu, installed apps, disk space usage etc.

If there are plans for more releases between now and getting a public listing on the Store (or at least signing packages with a "valid" cert ahead of that), would you consider generating a new cert with a narrower scope?  For comparison, Microsoft's WSL MSIX packages are signed using a cert with:

> **Enhanced Key Usage**
> Microsoft Publisher (1.3.6.1.4.1.311.76.8.1)
> Code Signing (1.3.6.1.5.5.7.3.3)

While a third-party package has:

> **Enhanced Key Usage**
> Code Signing (1.3.6.1.5.5.7.3.3)
> Windows Store (1.3.6.1.4.1.311.76.3.1)
> Unknown Key Usage (1.3.6.1.4.1.311.76.5.200)

I'm guessing only _Code Signing_ is needed here, and hopefully that can be added to _Trusted Publishers_ in the Windows certificate store.

**Comment by Chaphasilor** [comment]
If you can give me clear instructions on how to generate those certs, sure! At the moment this is more of a bandaid solution anyway so people curious about it can install it. Desktop support is still in alpha. But if there's an easy way to improve the certificate, then we can switch to that of course.

**Comment by Terrance** [comment]
Assuming you're building for Windows on Windows, [`New-SelfSignedCertificate` in PowerShell](https://learn.microsoft.com/en-us/powershell/module/pki/new-selfsignedcertificate) can seemingly do this in one step:

```powershell
New-SelfSignedCertificate -KeyUsage DigitalSignature -KeySpec Signature -Type CodeSigningCert -Subject Finamp -CertStoreLocation Cert:\CurrentUser\My
```

This produces a new self-signed certificate and private key (defaults to RSA 2048-bit, valid for one year) with Code Signing extended key usage, and stores it in the `Current User/Personal/Certificates` store.  You may also want `-KeyExportPolicy Exportable` if you need to use the private key outside of the Windows certificate store.

You can probably get OpenSSL to do this for you too, but it'll likely require some amount of faff with CSR configuration files to set the key usage.

**Comment by Chaphasilor** [comment]
@Terrance thanks for that command!  
Can you try if the new cert (`windows\runner\resources\FINAMP_MSIX_LOCAL_CERTIFICATE_NEW.pfx`) works with the latest artifact form the CI build here?: https://github.com/jmshrv/finamp/actions/runs/21334630308

If that works we can switch, no problems. One thing I'm not sure about is the support for beta versions on the Microsoft store, so we might need to stick with that cert for any beta releases? I doubt there's a way to get the private key form the Microsoft store cert.

**Comment by Terrance** [comment]
Looks like Windows is happy with it!

I can't extract the certificate from the encrypted PFX (a cursory search suggests the cert is also encrypted in this case), so I've just pulled the cert from the MSIX package signature info instead.

Apparently the cert still needs to be placed inside the _Third-Party Root Certification Authorities_ store rather than _Trusted Publishers_ in order to actually verify the MSIX, but I think with the scoped usage that's not a problem.

(Note that using Certificate Installer's "automatically-determined store" option to install the cert chooses _Intermediate Certification Authorities_ instead, which also doesn't work for validating the MSIX.)

<img width="400" alt="Installer" src="https://github.com/user-attachments/assets/abd548b0-93f9-4a77-b5ad-6914f3fe86c5" />  <img width="400" alt="Digital signature" src="https://github.com/user-attachments/assets/47e0c058-0d5e-4c28-bf46-efb6eb314a0c" />

**Comment by Chaphasilor** [comment]
Okay. In that case, would you mind opening a PR for updating the instructions in the new readme?  
https://github.com/jmshrv/finamp/tree/new-readme?tab=readme-ov-file#windows

**Comment by Terrance** [comment]
In the process of writing this up, I just found [some Microsoft self-signing docs](https://learn.microsoft.com/en-us/windows/msix/package/create-certificate-package-signing#import-the-certificate-to-the-local-machine-trusted-people-store) which use _Trusted People_ rather than a root CA cert store -- that does also work and seems like the more appropriate place, so I've gone with that.
